db/migrate/20150612135546_create_host_status.rb

```diff
@@ -11,7 +11,24 @@ def up
     add_column :hosts, :global_status, :integer, :default => 0, :null => false
 
     Host.includes(:host_statuses, :last_report_object).find_each do |host|
-      host.refresh_statuses
+      HostStatus.status_registry.each do |status_class|
+        status = host.get_status(status_class)
+        begin
+          status.refresh! if status.relevant?
+        rescue => e
+          # if the status is not ready to be saved because of missing migration
+          # we skip it and let it refresh by itself
+          logger.warn "skipping sub-status #{status.class} refresh, it's not ready"
+        end
+      end
+      host.host_statuses.reload
+      host.refresh_global_status
+
+      # validations could trigger dhcp/tftp orchestration, if smart proxy is unavailable
+      # migration would fail
+      unless host.save(:validate => false)
```

Comment:
Oh, that name is really confusing :-) but it would work so I'll incorporate it. I'll keep rescuing and validation skipping because of the other issue but this will even improve it. Thanks for good tip!